### ISSUE TITLE
AutoSmelt not working properly

### ISSUE THREAD

**Issue by mcwal18** [GC-spigot/AdvancedEnchantments#4070]
### Describe the bug

Minecraft server: Paper 1.20.1
Plugin version: 9.7.3

Default users without the OP the enchantmet AutoSmelt doesn't work. It doesn't smelt the ores/blocks.
But with the OP it works.

I have a ticket created in discord as well >> autosmelt-not-5885
Please it's been so long no one could help me!

Thanks!

### How to reproduce

N/A

### Screenshots / Videos

[ae_debug.txt](https://github.com/GC-spigot/AdvancedEnchantments/files/14535872/ae_debug.txt)
[autosmelt.txt](https://github.com/GC-spigot/AdvancedEnchantments/files/14535911/autosmelt.txt)

I have tried to put with SMELT @Block .. but it remains the same.

### "/ae plinfo" link

/ae plinfo skipPlugins

**Comment by GC-spigot** [comment]
@MasonX3 @mcwal18 Please provide `/ae plinfo`, it sounds like some plugin compatibility issue.

**Comment by MasonX3** [comment]
@GC-spigot here's mine: https://paste.md-5.net/ovojuhowal
I have another issue that I need to bring up with TP_DROPS, but I'll put that in another ticket.

**Comment by mcwal18** [comment]
> @MasonX3 @mcwal18 Please provide `/ae plinfo`, it sounds like some plugin compatibility issue.

Here is my one
https://paste.md-5.net/tonorimifi

**Comment by mcwal18** [comment]
> @GC-spigot here's mine: https://paste.md-5.net/ovojuhowal I have another issue that I need to bring up with TP_DROPS, but I'll put that in another ticket.

I found the bug, it seems is the NoCheatPlus anticheat.
I use this anticheat: https://builtbybit.com/resources/hq-custom-ncp-configuration-fork.475/

**Comment by MasonX3** [comment]
> > @GC-spigot here's mine: https://paste.md-5.net/ovojuhowal I have another issue that I need to bring up with TP_DROPS, but I'll put that in another ticket.
> 
> I found the bug, it seems is the NoCheatPlus anticheat. I use this anticheat: https://builtbybit.com/resources/hq-custom-ncp-configuration-fork.475/

@GC-spigot, I do not have NoCheatPlus, but I do have Vulcan.

**Comment by ThomasWega** [comment]
> I do not have NoCheatPlus, but I do have Vulcan.

Can you try without it please?

**Comment by MasonX3** [comment]
> > I do not have NoCheatPlus, but I do have Vulcan.
> 
> Can you try without it please?

I can, albeit very reluctantly since it worked fine before, only thing that's updated is AdvancedEnchantments.

**Comment by mcwal18** [comment]
> > > @GC-spigot here's mine: https://paste.md-5.net/ovojuhowal I have another issue that I need to bring up with TP_DROPS, but I'll put that in another ticket.
> > 
> > 
> > I found the bug, it seems is the NoCheatPlus anticheat. I use this anticheat: https://builtbybit.com/resources/hq-custom-ncp-configuration-fork.475/
> 
> @GC-spigot, I do not have NoCheatPlus, but I do have Vulcan.

So is can't be fixed?

**Comment by ThomasWega** [comment]
> I can, albeit very reluctantly since it worked fine before, only thing that's updated is AdvancedEnchantments.

Did that help?

**Comment by MasonX3** [comment]
> > I can, albeit very reluctantly since it worked fine before, only thing that's updated is AdvancedEnchantments.
> 
> Was that the issue? Does it work without the ac?

Apologies for the delay. With and without it still seems to not work... same goes for duplication glitches.

**Comment by MasonX3** [comment]
I also updated to the latest version.

**Comment by ThomasWega** [comment]
This is so weird honestly. But it's come conflict for sure. Good guess is the anticheats, but if that doesn't work that's weird. Some plugin needs to be blocking this interaction for some reason though. We don't even check for OP. Do you have any protection plugins on the server? Try without those

**Comment by MasonX3** [comment]
@ThomasWega @GC-spigot 

Source is now confirmed by me and my team, there is another report open labeled #4102.

This was tested on a barebones server: https://paste.md-5.net/ategetiqol

Vulcan, GriefDefender, and WorldGuard are on the Main server and the bug persisted when removed. 

To recreate the bug, you will need a few items:
1x Dispenser
1x Button 
1x Shulker Box (Empty or Filled)
1x Pickaxe with a SMELT enchantment, in our case it is called "Auto Smelt" or "Vitrify"
Level 100 in Mining for mcMMO

The way mcMMO calculates Double Drops is by flagging blocks that have already been placed by a player into RAM, not allowing double drops to be done twice. However, this only happens when the block is placed by a player. If the block is placed by a dispenser, it bypasses that RAM check and allows for Double Drops to be done again, and again, and again, and again.

Once you have the items above, simply place a Shulker into a Dispenser with the button, and activate it. At level 100, you have a 10% chance of Double Drops. If you would like a 100% Double Drop Rate, set your level to around 1000. Break the Shulker Box with your SMELT Pickaxe, and repeat.

**Comment by ThomasWega** [comment]
Is the dispenser thing something that only AE has issues with or McMMO as well?

**Comment by MasonX3** [comment]
> Is the dispenser thing something that only AE has issues with or McMMO as well?

Positive. We tested all of the blocks/entities that could be placed by a Dispenser and broken by an item with SMELT or TP_DROPS.
Tested this with the same pickaxe with SMELT/TP_DROPS, verified that the items duplicated after being placed by the dispenser, then disabled the relevant enchants and mined with the same pickaxe. No duplications occurred when the enchantments were removed from the equation.
